woocommerce/woocommerce-ios
# Use Result type for fetching a Note, Product, and ProductReview

Refs #2254. 

This is yet another small-ish dependency/improvement for the fix for #2254. I changed these methods to return `Result`:

-  [`NotificationsRemote.loadNotes`](https://github.com/woocommerce/woocommerce-ios/blob/c43f53cba1ecc601c8dd987f343bf20c2ed7e3e8/Networking/Networking/Remote/NotificationsRemote.swift#L15)
- [`ProductReviewsRemote.loadProductReview`](https://github.com/woocommerce/woocommerce-ios/blob/c43f53cba1ecc601c8dd987f343bf20c2ed7e3e8/Networking/Networking/Remote/ProductReviewsRemote.swift#L46)
- [`ProductsRemote.loadProduct`](https://github.com/woocommerce/woocommerce-ios/blob/c43f53cba1ecc601c8dd987f343bf20c2ed7e3e8/Networking/Networking/Remote/ProductsRemote.swift#L85)

They and their corresponding `Store` methods should be fairly covered by unit tests. I added a [new one for `ProductReviewStoreTests`](https://github.com/woocommerce/woocommerce-ios/blob/c43f53cba1ecc601c8dd987f343bf20c2ed7e3e8/Yosemite/YosemiteTests/Stores/ProductReviewStoreTests.swift#L239) which seems to be missing. 

### XCTestCase+Wait

As I mentioned in https://github.com/woocommerce/woocommerce-ios/pull/2261#discussion_r424067204, I made a wrong assumption about how `public` works for files in test targets (e.g. `NetworkingTests`). Apparently, you couldn't access those in YosemiteTests. I fixed that in c5cebeb and added a similar helper for YosemiteTests in 00701a4.

## Testing

I think the existing unit tests should fairly cover these. Please scrutinize the changes though. 🙂 

Please also do a regression test:

1. Navigate to Reviews
2. Pull to refresh.
3. Confirm that no errors happen. 
4. Tap on a review
5. Pull to refresh.
6. Confirm that no errors happen. 

## Reviewing 

Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
